### PR TITLE
Support memory pool for FFT functions

### PR DESCRIPTION
**Pull request by not522**
Original implementation of cupy.fft does not use memory pool at the stage of making a FFT plan.
It spoils the computation speed and has possibility to raise out of memory exception.

### CODE REVIEW

File: cupy/cuda/cufft.pyx [hunk 1]
```diff
@@ -59,16 +68,60 @@ cpdef inline check_result(int result):
...
-    result = cufftDestroy(plan)
-    check_result(result)
-
-
-cpdef plan1d(int nx, int type, int batch):
-    cdef Handle plan
-    result = cufftPlan1d(&plan, nx, <Type>type, batch)
-    check_result(result)
-    return plan
+class Plan1d(object):
```
Comment (okuta) [inline cupy/cuda/cufft.pyx]: `scikit.cuda` provides `Plan` calss. [link](http://scikit-cuda.readthedocs.io/en/latest/generated/skcuda.fft.Plan.html)
Which do you think is better?

Comment (not522) [reply cupy/cuda/cufft.pyx]: I think Plan1d is the better choice in the current implementation.
cuFFT supports 1D/2D/3D/Many plans, therefore we should clarify which one we are using.
If we support other plans or the general plan in the future, we will add or rename classes.

Comment (okuta) [reply cupy/cuda/cufft.pyx]: OK, I understand.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**